internal/handler/turn.go

```diff
@@ -58,15 +58,19 @@ func (h *Handler) GetTurnAuth(w http.ResponseWriter, r *http.Request, params typ
 		return
 	}
 
-	if len(servers) != 1 {
-		h.log.Info("multiple TURN servers available: generating credentials only for the first one")
+	// collect URIs
+	uris := []string{}
+	for _, s := range servers {
+		for _, uri := range *s.Urls {
+			uris = append(uris, uri)
+		}
 	}
 
 	turnAuthToken := types.TurnAuthenticationToken{
 		Username: servers[0].Username,
```

Comment:
I'm afraid this is more difficult than that. The problem is that the auth-service can generate the TURN/ICE credentials for multiple gateways in a single call (e.g., all gateways in a namespace, or all gateways in a cluster), and each can use a different auth realm with different users/passwords. 

The way this should be implemented is to collect only the URIs with the same credentials and return only those. But in any way this may still turn out to be wrong since we never know *which* particular gateway to return (unless the caller chooses one with the proper HTTP queries).

Reply:
Thanks for starting this discussion! I agree. I had to sleep on it and came up with the idea to combine iceServers that have the same (username, password, cluster) tuple. The same username and password are not sufficient since they are not unique (see our demo cluster for an example). Likewise, keeping only the cluster (e.g., a media server) is not unique either. We can define multiple gateways for a single cluster (I learned this the hard way with the "multiple configs" test case, which might be not correct in this PR). My next idea is to collect URIs sharing the same (username, password, cluster). Wdyt?

Reply:
Here is one thing we have to understand first. The only time when we must provide multiple URIs per TURN auth token is when a single Gateway has multiple listeners. Now the way this implemented is that the TURN auth request handler calls the [ICE server config generation routine](https://github.com/l7mp/stunner-auth-service/blob/0ffcae3d501cbd895cf651117ede67ea1e262bb4/internal/handler/ice.go#L101), which returns an ICE auth token with one URI per listener, i.e., it does what we want it to do (see this [test](https://github.com/l7mp/stunner-auth-service/blob/0ffcae3d501cbd895cf651117ede67ea1e262bb4/ice_test.go#L55)). Once the TURN handler [has the resultant ICE config](https://github.com/l7mp/stunner-auth-service/blob/0ffcae3d501cbd895cf651117ede67ea1e262bb4/internal/handler/turn.go#L37), it just [copies the URIs as is](https://github.com/l7mp/stunner-auth-service/blob/0ffcae3d501cbd895cf651117ede67ea1e262bb4/internal/handler/turn.go#L69). My understanding is that this should contain multiple URIs, so why this is not happening?

Reply:
You are correct. Having a single gateway with multiple listeners indeed results multiple URIs:
```console
{"password":"pass-1","ttl":86400,"uris":["turn:172.18.0.6:3478?transport=udp","turn:0.0.0.0:3478?transport=tcp"],"username":"user-1"}
```

I guess my misleading comes from one gateways per protocol approach (e.g., [simple-tunnel](https://github.com/l7mp/stunner/tree/main/docs/examples/simple-tunnel) example).